### PR TITLE
Feature/excludedgroups

### PR DESCRIPTION
**Pull request by shortsn**
Hey paket-team,

i really like paket and want to give something back. I tried to implement the feature requested in #1630 because i think it will be a useful extension.

### CODE REVIEW

File: src/Paket.Core/PackageMetaData.fs [hunk 1]
```diff
@@ -111,6 +111,16 @@ let addDependency (templateFile : TemplateFile) (dependency : PackageName * Vers
...
 
+let excludeDependency (templateFile : TemplateFile) (exclude : PackageName) = 
+    match templateFile with
+    | CompleteTemplate(core, opt) -> 
+        let newExcludes = 
+            opt.ExcludedDependencies |> Set.add exclude
+        { FileName = templateFile.FileName
+          Contents = CompleteInfo(core, { opt with ExcludedDependencies = newExcludes }) }
+    | IncompleteTemplate -> 
+        failwith "You should only try to exclude dependencies to template files with complete metadata."
```
Comment (forki) [inline src/Paket.Core/PackageMetaData.fs]: can you please extend this message to include the filename?

**Comment by forki** [comment]
wow cool stuff!

File: src/Paket.Core/PackageMetaData.fs (hunk 1 above)
Comment (shortsn) [inline src/Paket.Core/PackageMetaData.fs]: done

**Comment by shortsn** [comment]
Is there anything i can do to fix the build? I dont think that my changes broke it and the failed test "Paket.IntegrationTests.InstallSpecs.#1592" seems to be a problem in other active pull requests too.

**Comment by forki** [comment]
I think it's unrelated.

**Comment by forki** [comment]
thanks for this great addition

**Pull request merged**